### PR TITLE
CIRCSTORE-509: Create CRUD API for storing circulation settings

### PR DESCRIPTION
**Pull request by MagzhanArtykov**
Covers: [CIRCSTORE-509](https://folio-org.atlassian.net/browse/CIRCSTORE-509)

**Purpose**
Create a new API (GET, POST, PUT, DELETE) for storing circulation settings . It will be used for syncing ECS TLR setting (MODTLR-44) through consortia publications.

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -104,6 +104,44 @@
...
     },
+    {
+      "id": "circulation-settings-storage",
+      "version": "1.0",
+      "handlers": [
+        {
+          "methods": ["GET"],
+          "pathPattern": "/circulation-settings-storage/circulation-settings",
+          "permissionsRequired": [
+            "circulation-settings-storage.circulation-settings.collection.get"
```
Comment (OleksandrVidinieiev) [inline descriptors/ModuleDescriptor-template.json:115]:
```suggestion
            "circulation-storage.circulation-settings.collection.get"
```
Please rename permissions in other endpoints below as well

File: ramls/circulation-setting.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "title": "Circulation Setting Schema",
+  "description": "Circulation setting",
+  "type": "object",
+  "properties": {
+    "id": {
+      "description": "ID of the circulation setting",
+      "type": "string"
```
Comment (OleksandrVidinieiev) [inline ramls/circulation-setting.json:9]: Shouldn't it match UUID pattern?

File: ramls/circulation-settings-storage.raml [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          schema: circulation-setting
+      get:
+        responses:
+          200:
+            description: "Circulation setting successfully retreived"
+            body:
+              application/json:
+                type: circulation-setting
+          501:
+            description: "Not implemented yet"
```
Comment (OleksandrVidinieiev) [inline ramls/circulation-settings-storage.raml:79]: Why do we need 501? (here and in other endpoints below)

File: ramls/examples/circulation-settings.json [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  "circulationSettings": [
+    {
+      "id": "497f6eca-6276-4993-bfeb-53cbbbba6f09",
+      "name": "Sample settings",
+      "value": {
+        "org.folio.circulation.settings": "true"
+      }
+    }
+  ],
+  "totalRecords": 0
```
Comment (OleksandrVidinieiev) [inline ramls/examples/circulation-settings.json:11]:
```suggestion
  "totalRecords": 1
```

File: src/main/java/org/folio/rest/impl/CirculationSettingsAPI.java [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        .onComplete(asyncResultHandler);
+  }
+
+  @Override
+  public void deleteCirculationSettingsStorageCirculationSettingsByCirculationSettingsId(
+    String circulationSettingsId, String lang, Map<String, String> okapiHeaders,
+    Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+
+      new CirculationSettingsService(vertxContext, okapiHeaders).delete(circulationSettingsId)
+        .onComplete(asyncResultHandler);
```
Comment (OleksandrVidinieiev) [inline src/main/java/org/folio/rest/impl/CirculationSettingsAPI.java:64]:
```suggestion
      new CirculationSettingsService(vertxContext, okapiHeaders).delete(circulationSettingsId)
        .onComplete(asyncResultHandler);
```
```suggestion
      new CirculationSettingsService(vertxContext, okapiHeaders)
        .delete(circulationSettingsId)
        .onComplete(asyncResultHandler);
```

**Review by OleksandrVidinieiev — APPROVED** [review]

**Review by alexanderkurash — APPROVED** [review]

**Pull request merged**